### ISSUE TITLE
spansMetric.SpansMetric construct has a bug (Compute object is not supported)

### ISSUE THREAD

**Issue by TirushV** [hashicorp/terraform-cdk#3570]
### Expected Behavior

```
export interface SpansMetricConfig extends cdktf.TerraformMetaArguments {
    /**
    * The name of the span-based metric. This field can't be updated after creation.
    *
    * Docs at Terraform Registry: {@link https://registry.terraform.io/providers/datadog/datadog/3.38.0/docs/resources/spans_metric#name SpansMetric#name}
    */
    readonly name: string;
    /**
    * filter block
    *
    * Docs at Terraform Registry: {@link https://registry.terraform.io/providers/datadog/datadog/3.38.0/docs/resources/spans_metric#filter SpansMetric#filter}
    */
    readonly filter?: SpansMetricFilter;
    /**
    * group_by block
    *
    * Docs at Terraform Registry: {@link https://registry.terraform.io/providers/datadog/datadog/3.38.0/docs/resources/spans_metric#group_by SpansMetric#group_by}
    */
    readonly groupBy?: SpansMetricGroupBy[] | cdktf.IResolvable;
}
```

### Actual Behavior

```
export interface SpansMetricConfig extends cdktf.TerraformMetaArguments {
    /**
    * The name of the span-based metric. This field can't be updated after creation.
    *
    * Docs at Terraform Registry: {@link https://registry.terraform.io/providers/datadog/datadog/3.38.0/docs/resources/spans_metric#name SpansMetric#name}
    */
    readonly name: string;
    /**
    * filter block
    *
    * Docs at Terraform Registry: {@link https://registry.terraform.io/providers/datadog/datadog/3.38.0/docs/resources/spans_metric#filter SpansMetric#filter}
    */
    readonly filter?: SpansMetricFilter;
    /**
    * group_by block
    *
    * Docs at Terraform Registry: {@link https://registry.terraform.io/providers/datadog/datadog/3.38.0/docs/resources/spans_metric#group_by SpansMetric#group_by}
    */
    readonly groupBy?: SpansMetricGroupBy[] | cdktf.IResolvable;
    readonly compute: SpansMetricCompute;
}
```

### Steps to Reproduce

We need to add this line in order to support compute objects and attribute to create span metric with cdk terraform datadog.

```
readonly compute: SpansMetricCompute;
```

### Versions

```
language: typescript
cdktf: 0.20.3
terraform: Terraform v1.7.2
macos: Macbook m2 pro
```

### Possible Solutions
```
readonly compute: SpansMetricCompute;
```

If we had this line we can pass the compute attribute to create the span metric via cdk terraform datadog.

### Anything Else?

Please try solving this ASAP. Thank you

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

I tried adding that suggested line in my local but getting below error with and without
```
⠹  Processing
[2024-03-28T22:36:12.096] [ERROR] default - ╷
│ Error: Invalid Block
│
│   with datadog_spans_metric.stubio_timeTakenInSeconds_custom_metrics_stubio-timeTakenInSeconds_5AA15605,
│   on cdk.tf.json line 627, in resource.datadog_spans_metric.stubio_timeTakenInSeconds_custom_metrics_stubio-timeTakenInSeconds_5AA15605:
│  627:       }
│
│ Block compute must have a configuration value as the provider has marked it
│ as required
dev-Stubio  ╷
            │ Error: Invalid Block
            │
            │   with datadog_spans_metric.stubio_timeTakenInSeconds_custom_metrics_stubio-timeTakenInSeconds_5AA15605 (stubio_timeTakenInSeconds_custom_metrics/stubio-timeTakenInSeconds),
            │   on cdk.tf.json line 627, in resource.datadog_spans_metric.stubio_timeTakenInSeconds_custom_metrics_stubio-timeTakenInSeconds_5AA15605 (stubio_timeTakenInSeconds_custom_metrics/stubio-timeTakenInSeconds):
            │  627:       }
            │
            │ Block compute must have a configuration value as the provider has marked it
            │ as required
            ╵
```

**Comment by TirushV** [comment]
Hi Team, Any update?

**Comment by TirushV** [comment]
Hello @ansgarm , We are using terraform cdk to provision our monitoring setup. Appreciate the help here if there is any ETA or any suggestion regarding the above bug. Looking forward for the response. Thank you.

**Comment by ansgarm** [comment]
Hi @TirushV 👋

I seem to have found a fix (#3580) that will resolve this. In the meantime you could also use a [manual override](https://developer.hashicorp.com/terraform/cdktf/concepts/resources#escape-hatch) to set that property.

**Comment by ansgarm** [comment]
Closed via #3580
